#35610 Document PostSharp 2025.0. (#14)
postsharp/PostSharp.Documentation: conceptual/Introduction/WhatsNew/breaking-changes-20250.md

---
uid: breaking-changes-20250
title: "Breaking Changes in PostSharp 2025.0"
product: "postsharp"
categories: "PostSharp;AOP;Metaprogramming"
summary: "PostSharp 2025.0 introduces several breaking changes and deprecates platforms and features."
---
# Breaking Changes in PostSharp 2025.0

PostSharp 2025.0 contains a small number of breaking changes mainly related to platform upgrades and dependency security.

## PostSharp
* To build .NET Standard, .NET Core and .NET projects, .NET 8.0 SDK is required. Older SDKs are no longer supported.

> * PostSharp aspects required to use `[PSerializable]` instead of `[Serializable]` for aspects serialization. 

Reviewer: ```suggestion
* PostSharp aspects must now use `[PSerializable]` instead of `[Serializable]` for aspects serialization. 
```

>   Projects that still require using `[Serializable]` for aspects should explicitly set the `PostSharpBinaryFormatterAllowed` MSBuild property to `True`, but use of this option is unsupported.

Reviewer: ```suggestion
  Projects that still require using `[Serializable]` for aspects should explicitly set the `PostSharpBinaryFormatterAllowed` MSBuild property to `True`, but the use of this option is unsupported, which means it is not guaranteed to work.
```
